src/types/models/CommunityModel.jl

```diff
@@ -0,0 +1,209 @@
+"""
+$(TYPEDEF)
+
+A standardized structure used to package models that can easily be combined into
+a [`CommunityModel`](@ref).
+
+# Fields
+$(TYPEDFIELDS)
+
+# Assumptions
+1. Exchange reactions, all of which are idenitified in `exchange_reaction_ids`,
+   have the form: `A[external] ⟷ ∅` where `A` is a metabolite. No other
+   exchanges are allowed.
+2. The biomass reaction of a model produces a biomass metabolite called
+   `biomass_metabolite_id` with stoichiometry 1.
+3. There is only one biomass reaction in the model.
+"""
+Base.@kwdef mutable struct CommunityMember
+    "Name of model appended to intracellular reactions and metabolites."
+    id::String
+    "Abundance of organism."
+    abundance::Float64
+    "Underlying model."
+    model::AbstractMetabolicModel
+    "List of all exchange reactions in model."
+    exchange_reaction_ids::Vector{String}
+    "ID of biomass metabolite."
+    biomass_metabolite_id::String
+end
+
+"""
+$(TYPEDEF)
+
+# Fields
+$(TYPEDFIELDS)
+
+# Assumptions
+1. Each exchanged metabolite in the underlying `models`, identified through the
+   exchange reactions, will get an environmental variable.
+2. It is assumed that the same namespace is used to identify unique exchanged
+   metabolites.
+3. The objective created by this model is the equal growth rate/balanced growth
+   objective. In short, all biomass metabolites are produced at the same rate.
+"""
+Base.@kwdef mutable struct CommunityModel <: AbstractMetabolicModel
+    "Models making up the community."
+    members::Vector{CommunityMember}
+    "Name of the objective"
+    objective_id::String = "equal_growth_rates_biomass_function"
+    "Bounds enforced on the environmental exchanged metabolites."
+    env_met_flux_bounds::Dict{String,Tuple{Float64,Float64}} =
+        Dict{String,Tuple{Float64,Float64}}()
+end
+
+"""
+$(TYPEDSIGNATURES)
+
+Return the reactions in `cm`, which is a [`CommunityModel`](@ref). All reactions
+have the `id` of each respective underlying [`CommunityMember`](@ref) appended
+as a suffix. All exchange environmental reactions have `ENV_` added as a prefix
+followed by the metabolite id.
+"""
+function Accessors.reactions(cm::CommunityModel)
+    rxns = [rid * "#" * m.id for m in cm.members for rid in reactions(m.model)]
+    env_exs = ["ENV_" * env_met for env_met in get_env_mets(cm)]
```

Comment:
stringology should be parametrized (move the function into `constants` or something)